dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/WebMvcMetricsConfig.java

```diff
@@ -48,16 +49,210 @@
 import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerMapping;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
+import org.springframework.web.util.UrlPathHelper;
 
 /**
  * @author Luciano Fiandesio
  */
 @Configuration
 @Conditional(WebMvcMetricsConfig.WebMvcMetricsEnabledCondition.class)
 public class WebMvcMetricsConfig implements WebMvcConfigurer {
+  private static final String ATTRIBUTE_START_TIME =
+      WebMvcMetricsFilter.class.getName() + ".START_TIME";
+
+  /**
+   * Interface defining how to generate Tags to be associated with metrics for Spring MVC requests.
+   */
+  public interface WebMvcTagsProvider {
+    /**
+     * Provides tags to be associated with metrics for the given request.
+     *
+     * @param request the HTTP request
+     * @param response the HTTP response
+     * @param handler the handler for the request or {@code null} if the handler is not known
+     * @param exception the exception, if any, thrown during request handling
+     * @return tags to associate with metrics for the request
+     */
+    Iterable<Tag> getTags(
+        HttpServletRequest request,
+        HttpServletResponse response,
+        Object handler,
+        Throwable exception);
+  }
+
+  /** Default implementation of {@link WebMvcTagsProvider}. */
+  public static class DefaultWebMvcTagsProvider implements WebMvcTagsProvider {
+
+    private static final Tag URI_NOT_FOUND = Tag.of("uri", "NOT_FOUND");
+    private static final Tag URI_REDIRECTION = Tag.of("uri", "REDIRECTION");
+    private static final Tag URI_ROOT = Tag.of("uri", "root");
+    private static final Tag URI_UNKNOWN = Tag.of("uri", "UNKNOWN");
+    private static final Tag EXCEPTION_NONE = Tag.of("exception", "None");
+    private static final Tag STATUS_UNKNOWN = Tag.of("status", "UNKNOWN");
+
+    private final UrlPathHelper urlPathHelper = new UrlPathHelper();
+
+    @Override
+    public Iterable<Tag> getTags(
+        HttpServletRequest request,
+        HttpServletResponse response,
+        Object handler,
+        Throwable exception) {
+      return Tags.of(
```

Comment:
Is this code not running when the configuration is not enabled?
If not I guess we should skip this if we don't collect the metrics?

Reply:
No, this is only running if enabled.
This: @Conditional(WebMvcMetricsConfig.WebMvcMetricsEnabledCondition.class) enables/disables the whole class.